### PR TITLE
Update Onyx port, Add Derecho intel port

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Update Onyx port, Add Derecho intel port
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    All tests pass, ran full test suites run on Onyx  (3 compilers) and Derecho.  No baselines to compare to at this time.  https://github.com/CICE-Consortium/Test-Results/wiki/icepack_by_hash_forks#66de407dc88bd5a9189cd60ce31eb38deadd27c7, https://github.com/CICE-Consortium/Test-Results/wiki/icepack_by_hash_forks#df691e4ba1aa121e650ff69c93c4651ab9ef5850
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [ ] Please provide any additional information or relevant details below:

### CODE REVIEW

File: configuration/scripts/machines/env.onyx_cray [hunk 1]
```diff
@@ -39,17 +39,18 @@ module load craype-broadwell
...
 setenv ICE_MACHINE_MACHINFO "Cray XC40/50 Xeon E5-2699v4 Broadwell"
 setenv ICE_MACHINE_ENVNAME cray
-setenv ICE_MACHINE_ENVINFO "Cray cce/11.0.2"
+setenv ICE_MACHINE_ENVINFO "Cray cce/14.0.3, netcdf/4.8.1.3"
 setenv ICE_MACHINE_MAKE gmake
-setenv ICE_MACHINE_WKDIR $WORKDIR/ICEPACK_RUNS
+setenv ICE_MACHINE_WKDIR $WORKDIR/CICE_RUNS
 setenv ICE_MACHINE_INPUTDATA /p/app/unsupported/RASM/cice_consortium
-setenv ICE_MACHINE_BASELINE $WORKDIR/ICEPACK_BASELINE
+setenv ICE_MACHINE_BASELINE $WORKDIR/CICE_BASELINE
```
Comment (eclare108213) [inline configuration/scripts/machines/env.onyx_cray:53]: Should this be changing from ICEPACK to CICE?

File: configuration/scripts/machines/env.onyx_gnu [hunk 2]
```diff
@@ -39,17 +39,18 @@ module load craype-broadwell
...
 setenv ICE_MACHINE_MACHINFO "Cray XC40/50 Xeon E5-2699v4 Broadwell"
 setenv ICE_MACHINE_ENVNAME gnu
-setenv ICE_MACHINE_ENVINFO "GNU Fortran (GCC) 10.2.0"
+setenv ICE_MACHINE_ENVINFO "GNU Fortran (GCC) 12.1.0, netcdf/4.8.1.3"
 setenv ICE_MACHINE_MAKE gmake
-setenv ICE_MACHINE_WKDIR $WORKDIR/ICEPACK_RUNS
+setenv ICE_MACHINE_WKDIR $WORKDIR/CICE_RUNS
 setenv ICE_MACHINE_INPUTDATA /p/app/unsupported/RASM/cice_consortium
-setenv ICE_MACHINE_BASELINE $WORKDIR/ICEPACK_BASELINE
+setenv ICE_MACHINE_BASELINE $WORKDIR/CICE_BASELINE
```
Comment (eclare108213) [inline configuration/scripts/machines/env.onyx_gnu:53]: check CICE vs ICEPACK

File: configuration/scripts/machines/env.onyx_intel [hunk 3]
```diff
@@ -39,17 +39,18 @@ module load craype-broadwell
...
 setenv ICE_MACHINE_MACHINFO "Cray XC40/50 Xeon E5-2699v4 Broadwell"
 setenv ICE_MACHINE_ENVNAME intel
-setenv ICE_MACHINE_ENVINFO "ifort 19.1.3.304"
+setenv ICE_MACHINE_ENVINFO "ifort 2021.4.0, netcdf/4.8.1.3"
 setenv ICE_MACHINE_MAKE gmake
-setenv ICE_MACHINE_WKDIR $WORKDIR/ICEPACK_RUNS
+setenv ICE_MACHINE_WKDIR $WORKDIR/CICE_RUNS
 setenv ICE_MACHINE_INPUTDATA /p/app/unsupported/RASM/cice_consortium
-setenv ICE_MACHINE_BASELINE $WORKDIR/ICEPACK_BASELINE
+setenv ICE_MACHINE_BASELINE $WORKDIR/CICE_BASELINE
```
Comment (eclare108213) [inline configuration/scripts/machines/env.onyx_intel:53]: and again

**Comment by apcraig** [comment]
Good catch, fixed.

**Review by eclare108213 — APPROVED** [review]
approving based on visual inspection

**Pull request merged**